src/github.com/couchbase/sync_gateway/base/logging.go

```diff
@@ -221,10 +227,37 @@ func logWithCaller(color string, prefix string, format string, args ...interface
 	message := fmt.Sprintf(format, args...)
 	logLock.RLock()
 	defer logLock.RUnlock()
-	logger.Print(color, prefix, ": ", message, reset,
+	Print(color, prefix, ": ", message, reset,
 		dim, " -- ", GetCallersName(2), reset)
 }
 
+// Simple wrapper that converts Print to Printf
+func Print(args ...interface{}) {
+    ok := logLevel <= 1
+
+    if ok {
+        Printf("%s", fmt.Sprint(args...))
+    }
+}
+
+// Logs a formatted message to the underlying logger
+func Printf(format string, args ...interface{}) {
+    logLock.RLock()
+    defer logLock.RUnlock()
+    ok := logLevel <= 1
+
+    if ok {
+        if !logNoTime {
+            timestamp := time.Now().Format(timestampPattern)
+            var tsarg []interface{}
+            args = append(append(tsarg, timestamp), args...)
+            logger.Printf("%s "+format, args...)
```

Comment:
I'm wondering about the performance of lines 253 and 254 - the double append in 253, and using + instead of copy on 254.  What's your take - is there any chance of performance degradation after this change?  What do you think about writing a benchmark for our standard logging, to make sure we don't have any CPU or memory usage degradation?